File: lib/services/usuario_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../core/constants/api_constants.dart';
import '../models/usuario.dart';

class UsuarioService {
  static String get url => '${ApiConstants.baseUrl}/usuarios';

  static Future<List<Usuario>> obtenerUsuarios() async {
    final res = await http.get(Uri.parse(url));

    if (res.statusCode == 200) {
      final List data = jsonDecode(res.body);
      return data.map((e) => Usuario.fromJson(e)).toList();
    } else {
      throw Exception('Error al obtener usuarios');
    }
  }

  static Future<void> crearUsuario(
      String nombre, String correo, String password, String rol) async {
    final res = await http.post(
      Uri.parse(url),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'nombre': nombre,
        'correo': correo,
        'password': password,
        'rol': rol,
      }),
    );

    if (res.statusCode != 201) {
      throw Exception('Error al crear usuario');
    }
  }

  static Future<void> actualizarUsuario(
      String id, Usuario usuario) async {
    final res = await http.put(
      Uri.parse('$url/$id'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(usuario.toJson()),
    );

    if (res.statusCode != 200) {
      throw Exception('Error al actualizar usuario');
    }
  }

  static Future<void> eliminarUsuario(String id) async {
    final res = await http.delete(Uri.parse('$url/$id'));

    if (res.statusCode != 200) {
      throw Exception('Error al eliminar usuario');
    }
  }
}

File: src/services/usuario.service.js
const axios = require('axios');

const API_USUARIOS = process.env.USUARIOS_API_URL || 'http://127.0.0.1:8000/usuarios';

exports.obtenerUsuarios = async () => {
  const res = await axios.get(API_USUARIOS);
  return res.data;
};

exports.crearUsuario = async (usuario) => {
  const res = await axios.post(API_USUARIOS, usuario);
  return res.data;
};

exports.obtenerUsuario = async (id) => {
  const res = await axios.get(`${API_USUARIOS}/${id}`);
  return res.data;
};

exports.actualizarUsuario = async (id, usuario) => {
  const res = await axios.put(`${API_USUARIOS}/${id}`, usuario);
  return res.data;
};

// 🔥 ELIMINACIÓN LÓGICA (HU7)
exports.eliminarUsuario = async (id) => {
  const usuarioActual = await axios.get(`${API_USUARIOS}/${id}`);

  const usuarioActualizado = {
    ...usuarioActual.data,
    activo: false
  };

  const res = await axios.put(`${API_USUARIOS}/${id}`, usuarioActualizado);
  return res.data;
};